File: Lab 1/src/main/java/LexerWrapper.java
import lexer.Lexer;
import lexer.Token;

import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.List;

public class LexerWrapper {

    private List<Token> tokens;

    private String fileName;

    public LexerWrapper(String fileName) {
        this.fileName = fileName;
        Lexer lexer = new Lexer(fileName);
        lexer.tokenize();
        tokens = lexer.getTokens();
    }

    public void printTokens() {
        int i = 0;
        for (Token token : tokens) {
            System.out.println(++i + "   " + token.toString() + "\n");
        }
    }

    public void toHtml() {
        String head = "<html>\n" +
                "    <head>\n" +
                "        <link rel=\"stylesheet\" href=\"style.css\">\n" +
                "        <title>%s</title>\n" +
                "    </head>\n" +
                "<body>";

        String bum = "</body>\n" +
                "</html>";

        StringBuilder document = new StringBuilder();
        document.append(String.format(head, fileName));

        for (var token : tokens) {
            document.append(styleText(token));
        }
        document.append(bum);

        FileWriter fileWriter = null;
        try {
            fileWriter = new FileWriter("src/main/resources/" + fileName + ".html");
        } catch (IOException e) {
            e.printStackTrace();
        }
        PrintWriter printWriter = new PrintWriter(fileWriter);
        printWriter.print(document);
        printWriter.close();
    }

    private String styleText(Token token) {
        return "<span class=\"" + token.getType().toString().toLowerCase() + "\">"
                + token.getTokenString() + "</span>";
    }
}

File: Lab 1/src/main/java/LexerWrapper.kt
import com.hp.gagawa.java.elements.*
import lexer.Lexer
import lexer.Token
import java.io.File
import java.io.FileNotFoundException
import java.io.FileOutputStream
import java.io.PrintWriter
import java.util.stream.Collectors

class LexerWrapper(fileName: String) {
    private val tokens: List<Token>
    fun groupedTokens() {
        val mappedTokens = tokens.stream().collect(Collectors.groupingBy { obj: Token -> obj.type })
        mappedTokens.forEach { (key: Token.Type, value: List<Token>) ->
            if (key != Token.Type.WHITESPACE) {
                println("$key : ")
                for (`val` in value) {
                    println("\t" + `val`.tokenString)
                }
            }
        }
    }

    fun printTokens() {
        var i = 0
        for (token in tokens) {
            println((++i).toString() + "   " + token.toString() + "\n")
        }
    }

    private var out: PrintWriter? = null

    private var html: Html? = null
    private var body: Body? = null

    fun toHtml() {
        try {
            val file = File("./src/main/resources/index.html")
            out = PrintWriter(FileOutputStream(file))
            build()
            addTokens()
            finish()
        } catch (ex: FileNotFoundException) {
            ex.printStackTrace()
        }
    }

    private fun build() {
        html = Html()
        val head = Head()
        val title = Title().appendChild(Text("Reporting Page"))
        head.appendChild(title)
        val link = Link().setRel("stylesheet").setHref("style.css")
        head.appendChild(link)
        html!!.appendChild(head)
        body = Body()
    }

    private fun addTokens() {
        for (token in tokens) {
            val span = Span().setCSSClass(token.type.toString().toLowerCase()).appendText(token.tokenString)
            body!!.appendChild(span)
        }
    }

    private fun finish() {
        html!!.appendChild(body)
        out!!.println(html!!.write())
        out!!.close()
    }


    init {
        val lexer = Lexer(fileName)
        lexer.tokenize()
        tokens = lexer.tokens
    }
}